ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -2643,15 +2643,17 @@ private RelNode genJoinRelNode(RelNode leftRel, String leftTableAlias, RelNode r
             count++;
           }
           joinCond = count > 1 ? and : equal;
-        } else if (unparseTranslator != null && unparseTranslator.isEnabled()) {
-          genAllExprNodeDesc(joinCond, input, jCtx);
         }
         Map<ASTNode, RexNode> exprNodes = RexNodeTypeCheck.genExprNodeJoinCond(
             joinCond, jCtx, cluster.getRexBuilder());
         if (jCtx.getError() != null) {
           throw new SemanticException(SemanticAnalyzer.generateErrorMessage(jCtx.getErrorSrcNode(),
               jCtx.getError()));
         }
+        for (Map.Entry<ASTNode, RexNode> entry : exprNodes.entrySet()) {
+          UnparseTranslatorHelper.addTranslationIfNeeded(entry.getKey(), entry.getValue(),
```

Comment:
I think we are kinda approach a point where we should have the unparse translator enabled at all times...
I wonder how much we gain by keeping the support to have it disabled...